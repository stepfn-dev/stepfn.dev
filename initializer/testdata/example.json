{
  "StartAt": "First Unix date",
  "States": {
    "First Unix date": {
      "Type": "Task",
      "Resource": "arn:aws:states:::lambda:invoke",
      "Parameters": {
        "Payload.$": "$",
        "FunctionName": "unix"
      },
      "ResultSelector": {
        "First.$": "$.Payload"
      },
      "Next": "Second Unix date"
    },
    "Second Unix date": {
      "Type": "Task",
      "Resource": "arn:aws:states:::lambda:invoke",
      "Parameters": {
        "Payload.$": "$",
        "FunctionName": "unix"
      },
      "ResultPath": "$.Second",
      "Next": "Refactor"
    },
    "Refactor": {
      "Type": "Pass",
      "Parameters": {
        "First.$": "$.First",
        "Second.$": "$.Second.Payload"
      },
      "Next": "Sum them"
    },
    "Sum them": {
      "Type": "Task",
      "Resource": "arn:aws:states:::lambda:invoke",
      "Parameters": {
        "Payload.$": "$",
        "FunctionName": "sum"
      },
      "ResultSelector": {
        "Sum.$": "$.Payload"
      },
      "ResultPath": "$.Sum",
      "End": true
    }
  }
}
