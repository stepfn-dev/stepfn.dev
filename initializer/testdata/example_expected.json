{
  "StartAt": "First Unix date",
  "States": {
    "First Unix date": {
      "Type": "Task",
      "Resource": "arn:aws:states:::lambda:invoke",
      "Parameters": {
        "Payload.$": "$",
        "FunctionName": "funcarn",
        "ClientContext": "eyJjdXN0b20iOnsiSWQiOiJkeW5hbW9pZCIsIkhhbmRsZXIiOiJ1bml4In19"
      },
      "ResultSelector": {
        "First.$": "$.Payload"
      },
      "Next": "Second Unix date"
    },
    "Second Unix date": {
      "Type": "Task",
      "Resource": "arn:aws:states:::lambda:invoke",
      "Parameters": {
        "Payload.$": "$",
        "FunctionName": "funcarn",
        "ClientContext": "eyJjdXN0b20iOnsiSWQiOiJkeW5hbW9pZCIsIkhhbmRsZXIiOiJ1bml4In19"
      },
      "ResultPath": "$.Second",
      "Next": "Refactor"
    },
    "Refactor": {
      "Type": "Pass",
      "Parameters": {
        "First.$": "$.First",
        "Second.$": "$.Second.Payload"
      },
      "Next": "Sum them"
    },
    "Sum them": {
      "Type": "Task",
      "Resource": "arn:aws:states:::lambda:invoke",
      "Parameters": {
        "Payload.$": "$",
        "FunctionName": "funcarn",
        "ClientContext": "eyJjdXN0b20iOnsiSWQiOiJkeW5hbW9pZCIsIkhhbmRsZXIiOiJzdW0ifX0="
      },
      "ResultSelector": {
        "Sum.$": "$.Payload"
      },
      "ResultPath": "$.Sum",
      "End": true
    }
  }
}